gdal/ogr/ogrsf_frmts/amigocloud/ogramigoclouddatasource.cpp

```diff
@@ -271,7 +267,10 @@ int OGRAmigoCloudDataSource::Open( const char * pszFilename,
             papoLayers[nLayers ++] = new OGRAmigoCloudTableLayer(this, papszTables[i]);
         }
         CSLDestroy(papszTables);
-        bOverwrite = CPLTestBool(CPLGetConfigOption("OVERWRITE", "NO"));
+        // If OVERWRITE: YES, truncate the layer.
+        if( nLayers==1 && CPLTestBool(CPLGetConfigOption("OVERWRITE", "NO")) ) {
```

Comment:
this is super dangerous to truncate in open with a config option !
Imagine if someone does ogr2ogr with an amigocloud dataset as source too...
You could use an open option to do so, which would be specified with -doo OVERWRITE=YES in ogr2ogr (doo = destination open option)), and would only apply to the destination dataset.

Reply:
How do I read doo value in Open?

Reply:
In papszOpenOptionsIn as done above with other options

Reply:
Also I have this error, when I try using it:
./apps/ogr2ogr -doo OVERWRITE=yes -f AmigoCloud "AmigoCloud:15276 datasets=100500" ~/Downloads/Visits.geojson

Warning 1: -doo ignored when creating the output datasource.

Reply:
Declare it in https://github.com/amigocloud/gdal/blob/7b87b13e2673f8933e48c849278419a0d9641278/gdal/ogr/ogrsf_frmts/amigocloud/ogramigoclouddriver.cpp#L106

Reply:
This works!
ogr2ogr -update -doo OVERWRITE=yes -f AmigoCloud "AmigoCloud:15276 datasets=100500" ~/Downloads/Visits.geojson